### PR TITLE
unicode

### PR DESCRIPTION
**Pull request by petersteneteg**
Better error message when registering a processor
Fixes for unicode conversion.in filesystem

### CODE REVIEW

**Review by martinfalk — DISMISSED** [review]

File: include/inviwo/core/processors/processorutils.h [hunk 1]
```diff
@@ -111,6 +113,23 @@ std::unique_ptr<T> makeProcessor(ivec2 pos, Args&&... args) {
     return p;
 }
 
+/**
+ * @brief Find which module that registered a processor
+ * @param processor the processor to look for
+ * @param app the InviwoApplication needed to get the modules
+ * @return the InviwoModule that registered the processor or nullptr if not found
+ */
+IVW_CORE_API InviwoModule* getProcessorModule(Processor* processor, const InviwoApplication& app);
```
Comment (ResearchDaniel) [inline include/inviwo/core/processors/processorutils.h]: should be const Processor*
Implementation should check that processor is not null.

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]

File: include/inviwo/core/processors/processorutils.h (hunk 1 above)
Comment (petersteneteg) [reply include/inviwo/core/processors/processorutils.h]: fixed

File: src/core/util/stringconversion.cpp [hunk 2]
```diff
@@ -60,9 +60,20 @@ std::wstring toWstring(const std::string& str) {
...
     auto state = std::mbstate_t();
     auto sptr = str.data();
+    const char* loc = nullptr;
     size_t len = std::mbsrtowcs(nullptr, &sptr, 0, &state);
+    if (len == static_cast<std::size_t>(-1)) {
+        loc = std::setlocale(LC_CTYPE, nullptr);
+        std::setlocale(LC_CTYPE, "en_US.UTF-8");
+        len = std::mbsrtowcs(nullptr, &sptr, 0, &state);
+        if (len == static_cast<std::size_t>(-1)) {
+            if (loc) std::setlocale(LC_CTYPE, loc);
```
Comment (ResearchDaniel) [inline src/core/util/stringconversion.cpp:71]: Should be an else here for throwing the exception?

File: include/inviwo/core/processors/processorutils.h (hunk 1 above)
Comment (ResearchDaniel) [reply include/inviwo/core/processors/processorutils.h]: No changes pushed yet...

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]

File: src/core/util/stringconversion.cpp (hunk 2 above)
Comment (petersteneteg) [reply src/core/util/stringconversion.cpp:71]: no, I just want to restore the local before I leave by throwing

**Comment by petersteneteg** [comment]
@ResearchDaniel done

File: include/inviwo/core/processors/processorutils.h (hunk 1 above)
Comment (ResearchDaniel) [reply include/inviwo/core/processors/processorutils.h]: Still not changed

Comment (petersteneteg) [reply include/inviwo/core/processors/processorutils.h]: that was very strange, I had fixed it locally and had now changes according to git, but it still was not committed, I restarted VS and the changes was gone. seems VS did something strange...
Should be fixed now

**Review by ResearchDaniel — DISMISSED** [review]

**Review by ResearchDaniel — APPROVED** [review]

**Pull request merged**